src/mito2/src/sst/parquet/flat_format.rs

```diff
@@ -0,0 +1,306 @@
+    /// SST file schema.
+    arrow_schema: SchemaRef,
+    override_sequence: Option<SequenceNumber>,
+}
+
+impl FlatWriteFormat {
+    /// Creates a new helper.
+    #[allow(dead_code)]
+    pub(crate) fn new(metadata: RegionMetadataRef, options: &FlatSchemaOptions) -> FlatWriteFormat {
+        let arrow_schema = to_flat_sst_arrow_schema(&metadata, options);
+        FlatWriteFormat {
+            metadata,
+            arrow_schema,
+            override_sequence: None,
+        }
+    }
+
+    /// Set override sequence.
+    #[allow(dead_code)]
+    pub(crate) fn with_override_sequence(
+        mut self,
+        override_sequence: Option<SequenceNumber>,
+    ) -> Self {
+        self.override_sequence = override_sequence;
+        self
+    }
+
+    /// Gets the arrow schema to store in parquet.
+    #[allow(dead_code)]
+    pub(crate) fn arrow_schema(&self) -> &SchemaRef {
+        &self.arrow_schema
+    }
+
+    /// Convert `batch` to a arrow record batch to store in parquet.
+    #[allow(dead_code)]
+    pub(crate) fn convert_batch(&self, batch: &RecordBatch) -> Result<RecordBatch> {
+        debug_assert_eq!(batch.num_columns(), self.arrow_schema.fields().len());
+
+        let Some(override_sequence) = self.override_sequence else {
+            return Ok(batch.clone());
+        };
+
+        let mut columns = batch.columns().to_vec();
+        let sequence_array = Arc::new(UInt64Array::from(vec![override_sequence; batch.num_rows()]));
+        columns[sequence_column_index(batch.num_columns())] = sequence_array;
+
+        RecordBatch::try_new(self.arrow_schema.clone(), columns).context(NewRecordBatchSnafu)
+    }
+}
+
+/// Returns the position of the sequence column.
+pub(crate) fn sequence_column_index(num_columns: usize) -> usize {
+    num_columns - 2
+}
+
+// TODO(yingwen): Add an option to skip reading internal columns.
+/// Helper for reading the flat SST format with projection.
+///
+/// It only supports flat format that stores primary keys additionally.
+pub struct FlatReadFormat {
+    /// The metadata stored in the SST.
+    metadata: RegionMetadataRef,
+    /// SST file schema.
+    arrow_schema: SchemaRef,
+    /// Indices of columns to read from the SST. It contains all internal columns.
+    projection_indices: Vec<usize>,
+    /// Column id to their index in the projected schema (
+    /// the schema after projection).
+    column_id_to_projected_index: HashMap<ColumnId, usize>,
+    /// Column id to index in SST.
+    column_id_to_sst_index: HashMap<ColumnId, usize>,
+    /// Sequence number to override the sequence read from the SST.
+    override_sequence: Option<SequenceNumber>,
+}
+
+impl FlatReadFormat {
+    /// Creates a helper with existing `metadata` and `column_ids` to read.
+    ///
+    /// The given `arrow_schema` is the schema created from the `metadata`
+    /// by [to_flat_sst_arrow_schema()].
+    pub fn new(
+        metadata: RegionMetadataRef,
+        arrow_schema: SchemaRef,
```

Comment:
nit: Can we embed the creation of `arrow_schema` insde `FlatReadFormat::new` by invoking `to_flat_sst_arrow_schema`?